scipy/cluster/_hierarchy.pyx

```diff
@@ -160,7 +160,13 @@ cpdef void cluster_maxclust_monocrit(const double[:, :] Z, const double[:] MC, i
     max_nc : int
         The maximum number of clusters.
     """
-    cdef int i, k, i_lc, i_rc, root, nc, lower_idx, upper_idx
+    cdef int i
+    if max_nc >= n:
+        for i in range(n):
+            T[i] = i + 1
+        return
```

Comment:
This seems correct, right? If there is effectively no limit to the number of clusters we can ignore all thresholds and just assign each observation in its own cluster (sort of "by convention"). The numbering will be from 1 to n inclusive (convention in `fcluster`).

Reply:
Yes, I think so. Unless the points are not unique, perhaps. But I still think it's allowed by the description of the method:

> Finds a minimum threshold `r` so that the cophenetic distance between any two original observations in the same flat cluster is no more than `r` and no more than `t` flat clusters are formed.